client.go

```diff
@@ -296,21 +322,29 @@ func (client *Client) Execute(req *Request) (*Response, error) {
 
 	// convert a resty response into our Response object
 
-	//var err error
+	// var err error
 
 	resp = &Response{
-		//RestyResponse: restyResponse,
+		// RestyResponse: restyResponse,
 		Success:    restyResponse.IsSuccess(),
 		StatusCode: restyResponse.StatusCode(),
 		Status:     restyResponse.Status(),
 		ReceivedAt: restyResponse.ReceivedAt(),
 		Size:       restyResponse.Size(),
 		Body:       restyResponse.Body(), // byte[]
+		Error:      err,
 	}
 
 	// determine success and set err accordingly
 	if !resp.Success {
-		err = fmt.Errorf("API returned HTTP %d", resp.StatusCode)
+		var certErr x509.UnknownAuthorityError
+		if errors.As(resp.Error, &certErr) {
+			err = fmt.Errorf("%w, %s", certErr, sslCertErrorMsg)
```

Comment:
If it is possible, we should return the raw cert error so that the provider can detect it.